grafana/grafana
# InfluxDB: InfluxQL query editor: better handling of slow queries

in the influxql query-editor, when you open a dropdown that is filled with options loaded from the server (for example, you want to add a section to the `WHERE` part, we show a select-box, but first we load the options from the server. if the server is responding slowly, for a while the select-box will say "no options found", while loading in the options.

this PR changes it so that (in a somewhat hackish way), we set the "no items found" message to "loading" while we are loading. so, while the items are loading you see a "Loading" message in the select-box.

please note this does not apply to the select-box for the measurement-dropdown. that one uses a different style of select-box.

how to test:
- you need a slow influxdb data source 😁 we can use the `slow_proxy` or the `slow_proxy_mac` data sources (in this example i will use `slow_proxy_mac`
- `make devenv sources=influxdb,slow_proxy_mac ORIGIN_SERVER=http://host.docker.internal:8086 SLEEP_DURATION=4s`
- create a new influxdb data source, basically copy over everything from the `gdev-influxdb-influxql` data source (the Authorization header is set to `Token mytoken` (see https://github.com/grafana/grafana/blob/main/devenv/datasources.yaml#L69) , except change the server-url to `http://localhost:3011`
- you can use this data-source to test the loading-message.
